### PR TITLE
feat: editor initial test suite

### PR DESCRIPTION
**Pull request by aoltean-plenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/__tests__/support/pageObjects/EditorObject.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  get editorTextarea() {
+    return cy.getByTestId('editor-textarea');
+  }
+
+  get exitEditorButton() {
+    return cy.get('#close')
+  }
+
+  togglePreviewMode() {
+    this.editPreviewButton.should('be.enabled').click({ force: true });
```
Comment (ksted) [inline apps/web/__tests__/support/pageObjects/EditorObject.ts]: **question**: Is `force: true` really necessary?

File: apps/web/__tests__/support/pageObjects/EditorObject.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    this.editPreviewButton.should('contain.text', 'Preview');
+    return this;
+  }
+  toggleEditMode() {
+    this.editPreviewButton.should('be.enabled').click({ force: true });
```
Comment (ksted) [inline apps/web/__tests__/support/pageObjects/EditorObject.ts]: **question**: Is `force: true` really necessary?

File: apps/web/__tests__/support/pageObjects/EditorObject.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        } else {
+          cy.wrap($el).clear();
+        }
+      })
+      .type(content, {delay: 0})
+      .should('have.value', content);
+  }
+
+  checkEditorChanges() {
+    this.exitEditorButton.get('#close').click({ force: true });
```
Comment (ksted) [inline apps/web/__tests__/support/pageObjects/EditorObject.ts:109]: **question**: Is `force: true` really necessary?

**question**: `exitEditorButton` already does a `cy.get('#close')`. Why does this need `get('#close')` again?

File: apps/web/__tests__/test/editor/editMode.cy.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  it('should display and interact with the editor textarea', () => {
+    editor.isToolbarVisible();
+    editor.togglePreviewMode();
+    editor.toggleEditMode();
+    editor.isEditBlockVisible();
+    editor.checkEditorContent();
+
+    editor.replaceEditorContent(JSON.stringify(newContent, null, 2));
+    editor.checkEditorChanges();
```
Comment (ksted) [inline apps/web/__tests__/test/editor/editMode.cy.ts:20]: Good use of domain language. 👍

File: apps/web/__tests__/support/pageObjects/EditorObject.ts (hunk 3 above)
Comment (aoltean-plenty) [reply apps/web/__tests__/support/pageObjects/EditorObject.ts:109]: Hei! in the case of the close button I can't get the button via testid for dear life. The only thing that works is adding an id on the button.

**Review by ksted — APPROVED** [review]

**Pull request merged**